progress/README.md

```diff
@@ -0,0 +1,256 @@
+# Progress
+
+## **Requests**
+
+|                  Route                   |                         Description                          |
+| :--------------------------------------: | :----------------------------------------------------------: |
+|        [GET /progress](#get-post)        | Retrieves progress entries based on the provided parameters. |
+|     [POST /progress](#post-progress)     |                Creates a new progress entry.                 |
+| [DELETE /progress/:id](#delete-progress) |       Deletes a progress entry with the specified ID.        |
+
+## **GET /progress**
+
+Retrieves progress entries based on the provided parameters.
+
+- **Params**
+  None
+- **Query**  
+  type (query parameter): Specifies the type of progress to filter (e.g., "task", "user").
+  id (query parameter): Specifies the ID of the progress entry to retrieve (optional).
```

Comment:
I think we need to find a better way to represent query parameters. I think adding a `query parameter` is unnecessary and adding each query param in code blocks will have a better readability. For reference check other API contracts

Reply:
yeah Same was suggested by Ritik. Planning to add both these params as path params as both are always needed, what do you think about this approach.

Reply:
Yes I think that is a good idea